gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java

```diff
@@ -1390,6 +1445,40 @@ protected void configureRestModelGeneration(Project project, SourceSet sourceSet
     });
   }
 
+  protected void configurePegasusSchemaSnapshotGeneration(Project project, SourceSet sourceSet, boolean isExtensionSchema)
+  {
+    File schemaDir = isExtensionSchema? project.file(getExtensionSchemaPath(project, sourceSet))
+        : project.file(getDataSchemaPath(project, sourceSet));
+    if (isExtensionSchema && !SharedFileUtils.getSuffixedFiles(project, schemaDir, PDL_FILE_SUFFIX).isEmpty())
+    {
+      project.getLogger().debug("No extension schemas, skip generatePegasusExtensionSchemaSnapshot task");
+      return;
+    }
```

Comment:
At first glance, it's not obvious to me why this check only applies to the extension task.

Separately, there is a way to describe this behavior more declaratively that you can consider. Your implementation is pretty close to it already.

If you create a [FileTree](https://docs.gradle.org/current/userguide/working_with_files.html#sec:file_trees) that includes only PDL files, it can be passed to tasks that require them through a task property of type `Iterable<File>` (or a subclass, such as `FileTree` itself) and annotated with `@InputFiles`. Furthermore, the property can be annotated with `@SkipWhenEmpty`, which will cause Gradle to skip the task if no input files were found.

The benefit of this approach is that the task will always exist, so a script that runs, for instance, `./gradlew generatePegasusExtensionSchemaSnapshot` won't fail when no PDL files are found.

Reply:
I was thinking that extension task will run by default and extension schemas can only be PDL format, so I added this check.  The regular schema task only runs when configuration property is provided.

Updated GeneratePegasusSnapshotTask based on the suggestion.